lib/rules/no-useless-escape.js

```diff
@@ -148,16 +208,32 @@ module.exports = {
                     return;
                 }
 
-                nodeEscapes = VALID_STRING_ESCAPES;
+                let match;
+
+                while ((match = pattern.exec(value))) {
+                    validateString(VALID_STRING_ESCAPES, node, match);
+                }
             } else if (node.regex) {
-                nodeEscapes = VALID_REGEX_ESCAPES;
-            } else {
-                return;
-            }
+                parseRegExp(node.raw.slice(1, -1))
```

Comment:
Oops, I found a problem: `.slice(1, -1)` will be incorrect if the regex has flags, since the trailing slash won't be at index -1.